plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastoreConfig.java

```diff
@@ -17,22 +17,25 @@
 import io.airlift.configuration.ConfigDescription;
 import io.airlift.configuration.ConfigSecuritySensitive;
 import io.airlift.configuration.DefunctConfig;
-import jakarta.annotation.PostConstruct;
 import jakarta.validation.constraints.Max;
 import jakarta.validation.constraints.Min;
 
 import java.util.Optional;
 
-import static com.google.common.base.Preconditions.checkState;
-
-@DefunctConfig("hive.metastore.glue.use-instance-credentials")
+@DefunctConfig({
+        "hive.metastore.glue.use-instance-credentials",
+        "hive.metastore.glue.get-partition-threads",
+        "hive.metastore.glue.read-statistics-threads",
+        "hive.metastore.glue.write-statistics-threads",
+        "hive.metastore.glue.proxy-api-id",
+        "hive.metastore.glue.aws-credentials-provider",
```

Comment:
This PR removed the ability to specify a custom AWS credentials provider for Glue. Is that intentional @dain @findepi? Custom provider is useful for working around some corner cases, like #15267, which still applies to AWS SDK v2. Can we get a replacement for this similar to what's being done for S3 in #22162 and #22163?

Reply:
sounds good to me

Reply:
Opened #22425 to address this.